CMR/python/cmr/auth/token.py

```diff
@@ -0,0 +1,320 @@
+        file = open(path, "r")
+        text = file.read().strip()
+        file.close()
+    else:
+        text = None
+    return text
+
+def _write_file(path, text):
+    """
+    Write (creating if need be) file and set it's content
+    Parameters:
+        path (string): path to file to write
+        text (string): content for file
+    """
+    path = os.path.expanduser(path)
+    cache = open(path, "w+")
+    cache.write(text)
+    cache.close()
+
+def _write_token_file(token_text, options=None):
+    """
+    Write out a token cache file
+    Parameters:
+        token_text (string) : content to write out
+        options (dictionary): overrides, responds to 'cmr.token.file'
+    """
+    if options is None:
+        options = {}
+    token_path = c.dict_or_default(options, "cmr_token_file", "~/.cmr_token")
+    _write_file(token_path, token_text)
+
+def _days_old_from_time(past_timestamp, current_timestamp=time.time(), age=1.0):
+    """
+    tests if a difference between timestamps are old
+    Parameters:
+        past_timestamp(float): timestamp of file
+        current_timestamp(float): default to now, timestamp
+        age(float): defaults to 1.0, days old
+    Returns:
+        true if the file is older then age
+    """
+    days_old = (current_timestamp - past_timestamp)/60.0/60.0/24.0
+    return days_old > age
+
+def _old_file(path):
+    """
+    non-pure function
+    to get age of a file from today
+    Parameters:
+        path(string): path to file to test
+    Returns:
+        true if the file is older then one day
+    """
+    return _days_old_from_time(os.path.getmtime(path))
+
+def _read_token_file(options=None):
+    """
+    load a password from a local user file assuming ~/.password
+    Parameters:
+        options(dictionary): responds to 'cmr.token.file'
+    Returns:
+        content of token file, None if it does not exist
+    """
+    if options is None:
+        options = {}
+    raw_path = c.dict_or_default(options, "cmr.token.file", "~/.cmr_token")
+    path = os.path.expanduser(raw_path)
+    clear_text = None if _old_file(path) else _read_file(path)
+    return clear_text
+
+def _execute_command(cmd):
+    """
+    a utility method to execute a shell command and return a string of the output
+    Parameters:
+        cmd(string) unix command to execute
+    Returns:
+        response from command
+    """
+    result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+    return result.stdout.decode('utf-8')
+
+# ##############################################################################
+# lambdas
+
+# All password lambda functions accept two parameters and return a string
+# Parameters:
+#   user_id(string): Earth Data Login user name
+#   options(dictionary): configuration object which may be used by the lambda
+# Returns:
+#   password
+
+def password(clear_password):
+    """
+    create a pass through lambda function to allow plain text passwords, not an
```

Comment:
Capitalization for first word in all documentation strings generally

Reply:
I did this on a lot of them, think I got them all.